### ISSUE TITLE
Flashing shadows

### ISSUE THREAD

**Issue by SebastJava** [linuxmint/muffin#653]
```
 * Muffin version: mutter 5.6.2
 * Distribution: Mint 21.1
 * Graphics:
  Device-1: VMware SVGA II Adapter driver: vmwgfx v: 2.19.0.0
  Display: x11 server: X.Org v: 1.21.1.3 driver: X: loaded: vmware
    unloaded: fbdev,modesetting,vesa gpu: vmwgfx resolution: 1440x900~60Hz
  OpenGL: renderer: SVGA3D; build: RELEASE; LLVM; v: 2.1 Mesa 22.0.5
 ```

**This video capture describes it all:**

https://user-images.githubusercontent.com/31940713/179120823-59a0be09-17bd-42cb-b07e-475eb5701e77.mp4

**Other information**
This is a duplicate from https://github.com/linuxmint/mint21-beta/issues/35  
This mint-21-beta repository being archived, i opened this new issue here, hoping this bug is not forgotten. This bug ruins my whole user experience... Sorry to insist.

**Comment by SebastJava** [comment]
I am now onto a newly acquired, refurbished, 100% clean powerful desktop: [inxi-b.txt](https://github.com/linuxmint/muffin/files/10404854/inxi-b.txt)

And i am on a fresh LinuxMint Cinnamon 21.1, all up-to-date. And i still get those flashing shadows. It is my only problem. But it is a big one, it ruins my UX, it just doesn't look polished, professional.

* Is this the right repository for this issue ?
* Is anybody working on this ?
* Is anybody reading this ?

This bug might not be so obvious at first, if you try to reproduce it in simple "laboratory" conditions. But as soon as you set a light background with some light GTK theme like Mint-Y, and then start using it in real life conditions, that is, having multiple windows overlapping each other, then you quickly get those "flashing" shadows. Sometimes it can be a subtle instability, and sometimes it gets strong, obvious.

**Comment by mtwebster** [comment]
Yes this is the right repository. Yes quite a bit of time has been spent on it - just because no one's responded doesn't mean it's not being worked on. 

I'm sorry your experience has been ruined. Please, try to be strong.

This is not as simple an issue as it may appear.

**Comment by SebastJava** [comment]
@mtwebster 
 
 > This is not as simple an issue as it may appear.
 
Yes, i had a feeling this was a complex thing. Probably hard to solve. That's good to know you are still working on it. That's all i wanted to know. The beta repository being archived, i wasn't sure... Thanks for replying.

P.S.: There is also this other, similar issue: https://github.com/linuxmint/muffin/issues/654

**Comment by mtwebster** [comment]
Cinnamon 5.6.7 should include a fix for this, though it potentially made #654 worse (at least for some) - let me know your results please.

**Comment by SebastJava** [comment]
@mtwebster

Still the same bugs, no difference... until i restart the computer ! After this computer restart it looks... PERFECT !

Well, that's after just a short, 60 seconds of testings. I am currently busy and i should come back on this later. Don't worry, this is quite interesting to me and i want to do some more tests within the next 48 hours.

* No flashing on session quit dialog. Tried it 1, 2, 3 times, to be sure, as the results can be different on 2nd or 3rd test.
* No cropped shadows either. Having multiple windows and shadows on top of each other also looks perfect: it all blends smoothly.
* The shadow is stronger when the window is focused and this is perfectly normal. I think that's a feature from the Mint-Y GTK theme's CSS...

Those quick, first tests were done on my newly acquired, refurbished computer. My SSDs and HDD are either new or renewed, with a fresh LinuxMint 21.1 installation on it. Interestingly, my new computer and the old one are both on an Intel Core i5 CPU with integrated graphics, but the old one was of 2nd generation while the new one is of 6th generation. More info: [HP EliteDesk 800 G2](https://github.com/linuxmint/muffin/files/10429681/inxi-Fxxxrz.txt) (All up-to-date and restarted)

Tested with the Mint-Y-Aqua standard GTK theme on a pure white background for 100% visibility:

Right-click on an empty area on the desktop > Change Desktop Background > Settings:
Picture aspect = No picture, Background color = Solid white

Thanks a lot.

P.S.: Please don't close this issue so fast. It looks like both #653 and #654 are solved but i want to spend some more time on testing...

**Comment by SebastJava** [comment]
I found 3 different shadow intensities:
1. subtle on unfocused windows
2. more visible on focused windows
3. strong when hovering the grouped window list's thumbnail above the panel, for an already focused window

But this doesn't look like a bug. Maybe i just haven't noticed before. Anyway, it doesn't look bad at all.

**Comment by mtwebster** [comment]
The grouped window list one is the 'hover peek' feature it's actually showing a clone of the window.  If the window's already visible it'll end up doubling over the shadow.

**Comment by SebastJava** [comment]
### NO Flashing shadows

- [x] No flashing shadows when hovering the session quit dialog. Tried it 1, 2, 3 times, to be sure, as the results can be different on 2nd or 3rd test.
- [x] Same test, but using Mint-X.
- [x] Same test, but back on Mint-Y-Aqua, and on the Software source apt refresh pop-up.

Tested using the Mint-Y-Aqua GTK theme with a pure solid white background. On [HP EliteDesk 800 G2](https://github.com/linuxmint/muffin/files/10429681/inxi-Fxxxrz.txt) All up-to-date and restarted.

ALL OK on both #653 and #654.

**Comment by clefebvre** [comment]
https://github.com/linuxmint/cinnamon/commit/d6c8697a3c46541857e84f978e7266d5267ab58a
